### PR TITLE
Enable python 3.11 support

### LINKED ISSUE #293: Dependencies do not support Python 3.11
https://github.com/SheffieldML/GPy/issues/998

Currently we've worked around this by pinning the floating python version for the `mlos_core` conda env below 3.11:
https://github.com/microsoft/MLOS/blob/1cd609263f5e84008cb75e09b4c8d8355bf84185/conda-envs/mlos_core.yml#L25

Options could be to drop GPy support (which is kinda slow anyways), but Emukit currently relies on it.

When we have SMAC support (#294) we could just drop Emukit as well.

### PR DESCRIPTION
**Pull request by bpkroth**
Closes #293

### CODE REVIEW

**Comment by bpkroth** [comment]
Hmm, seems pyrfr is still lacking 3.11 support: https://github.com/automl/random_forest_run/issues/75

**Comment by bpkroth** [comment]
Or maybe not.  Seems to pass in the python 3.11 conda check.  Just the devcontainer fails.  Will need to debug further later.

File: .github/workflows/build-dist-test.ps1
Comment (bpkroth) [inline .github/workflows/build-dist-test.ps1:1]: CLRF auto conversions.

Basically just removed the GPy install portion.

File: conda-envs/mlos-3.11.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@ dependencies:
   - nb_conda_kernels
   - matplotlib
   - seaborn
+  - swig
```
Comment (bpkroth) [inline conda-envs/mlos-3.11.yml:20]: swig is necessary for pyrfr in 3.11

File: mlos_bench/setup.py [hunk 2]
```diff
@@ -76,7 +76,7 @@
         'mlos-core==' + _VERSION,
         'requests',
         'json5',
-        'jsonschema',
+        'jsonschema<4.18.0',    # FIXME: temporarily pinning to avoid breaking changes in 4.18.0
```
Comment (bpkroth) [inline mlos_bench/setup.py:79]: This was causing CI issues due to the recently updated 4.18 release of jsonschema.  For now I pinned it back and will address it in #437

Comment (bpkroth) [reply mlos_bench/setup.py]: PR for this is in #438 which should get merged first.

**Comment by bpkroth** [comment]
#438 should get merged first and then this is ready

**Review by motus — APPROVED** [review]

**Pull request merged**